process/process.go

```diff
@@ -61,34 +61,102 @@ func (ps *processService) ProcessCreate(
 }
 
 type process struct {
-	pid      uint32                  // process id
-	root     string                  // root dir
-	cwd      string                  // current working dir
-	uid      uint32                  // effective uid
-	gid      uint32                  // effective gid
-	sgid     []int                   // supplementary groups
-	cap      cap.Capabilities        // process capabilities
-	status   map[string]string       // process status fields
-	nsInodes map[string]domain.Inode // process namespace inodes
-	ps       *processService         // pointer to parent processService
+	pid         uint32                  // process id
+	root        string                  // root dir
+	hroot       string                  // root dir from host perspective
+	cwd         string                  // current working dir
+	hcwd        string                  // current working dir from host perspective
+	uid         uint32                  // effective uid
+	gid         uint32                  // effective gid
+	sgid        []int                   // supplementary groups
+	cap         cap.Capabilities        // process capabilities
+	status      map[string]string       // process status fields
+	nsInodes    map[string]domain.Inode // process namespace inodes
+	initialized bool                    // process initialization completed
+	ps          *processService         // pointer to parent processService
 }
 
 func (p *process) Pid() uint32 {
+
+	if !p.initialized {
+		p.getInfo()
```

Comment:
It looks like p.getInf() should be renamed p.init() ? Because it appears it's initializing the process struct.

Reply:
Yep, agree, will change.